R/utilities-task.R

```diff
@@ -426,3 +426,77 @@ checkTaskInputsExist <- function(task) {
   validateIsOfType(task, "Task")
   return(sapply(task$getInputs(), file.exists))
 }
+
+
+#' @title loadUserDefinedTask
+#' @description
+#' Upload a user-defined function as workflow task
+#' @param workflow `Workflow` object or derived class
+#' @param taskFunction Function to be called by `Task` object
+#' @param taskName character name of task to initialize reporting of task
+#' @param active logical to set if task is run by workflow
+#' @param settings list of input arguments that can be used by `settings` input argument of `taskFunction`
+#' @param taskFunctionUsesAllSimulationSets logical if `taskFunction` requires all simulation sets as input, 
+#' `PopulationPlotTask` object is preferred to `PlotTask` object.
+#' @return Updated `Workflow` object
+#' @export
+loadUserDefinedTask <- function(workflow,
+                                taskFunction,
+                                taskName = "userDefinedTask",
+                                active = TRUE,
+                                settings = NULL,
+                                taskFunctionUsesAllSimulationSets = FALSE) {

```

Comment:
I wonder if this argument (taskFunctionUsesAllSimulationSets ) is really required.
The decision (**PlotTask** vs. **PopulationPlotTask**) could be done based on the type of the workflow: 
- In case of **MeanModelWorkflow**: Create **PlotTask**
- In case of **PopulationWorkflow**: Create **PopulationPlotTask**

Reply:
It depends on what kind of task the user intends to do.
As a first implementation of this wrapper, I think your suggestion simplifies a lot, which is better for the users.